apps/fxc-front/src/app/logic/score/scorer.ts

```diff
@@ -1,19 +1,98 @@
-import { CircuitType } from '@flyxc/optimizer/lib/api';
+import type { LatLonAltTime, ScoringResult } from '@flyxc/optimizer/lib/optimizer';
 
-export class Score {
-  distanceM: number;
-  indexes: number[];
-  multiplier: number;
-  circuit: CircuitType;
-  closingRadiusKm: number;
-  points: number;
+import { type Response as WorkerResponse } from '../../workers/optimizer';
+import ScoringWorker from '../../workers/optimizer?worker';
+import { getScoringRuleName, type LeagueCode } from './league/leagues';
 
-  constructor(score: Partial<Score>) {
-    this.distanceM = score.distanceM ?? 0;
-    this.indexes = score.indexes ?? [];
-    this.multiplier = score.multiplier ?? 1;
-    this.circuit = score.circuit ?? CircuitType.OpenDistance;
-    this.closingRadiusKm = score.closingRadiusKm ?? 0;
-    this.points = score.points ?? 0;
+export type ScoringResultHandler = (result: ScoringResult) => void;
+
+export class Scorer {
+  private scoringWorker?: Worker;
+  private currentScoringRequestId = 0;
+  /**
+   * Associate the result handler of every scoring request with the scoring request identifier.
+   * The scoring request identifier is a sequence number computed by incrementation of `currentScoringRequestId` field.
+   * This identifier is sent to the worker (`id` field of the message) and returned by the worker in the response
+   * message (`data.id` field).
+   * The handler is retrieved and invoked when the worker response is received.
+   */
+  private handlers: Map<number, ScoringResultHandler> = new Map();
+
+  /**
+   * Scores a track
+   *
+   * `handleScoringResult` function is invoked in a worker context which means that 'this' keyword is a reference
+   * to the worker itself. If a function body uses 'this' keyword  and is sent to the constructor as a reference,
+   * it will not work. In this case, the function should be wrapped in an arrow function (see example bellow).<br/>
+   *
+   * E.g:
+   * ```
+   * class MyClass {
+   * ...
+   *   handleResult(result: ScoringResult){
+   *   ...
+   *     this.doSomethingWithResult(result);
+   *   ...
+   *   }
+   *   ...
+   *   // although this is a valid code, it will not work because 'this' is the worker.
+   *   const scorer = new Scorer();
+   *   const scoringRequestId = scorer.score(track, league, this.handleResult);
+   *   // the correct syntax is:
+   *   const scoringRequestId = scorer.core(track, league, (result)=>this.handleResult(result));
+   * }
+   * ```
+   *
+   * @param track
+   * @param league
+   * @param handleScoringResult {ScoringResultHandler}
+   *        Takes the ScoringResult into account
+   * @return {number} a number that identifies this scoring request
+   */
```

Comment:
```suggestion
/**
 * Scores a track
 *
 * @param track
 * @param league
 * @param handleScoringResult Callback called with the scoring solution
 * @return  a number that identifies this scoring request
 */
```

This can be simplified